pleskdistup/actions/common_checks.py

```diff
@@ -94,6 +98,170 @@ def _do_check(self) -> bool:
+            ", ".join([outdated_php_packages[installed] for installed in installed_pkgs]),
+            " ".join(outdated_php_packages[installed].replace(" ", "") for installed in installed_pkgs).lower())
+
+        log.debug("Outdated PHP versions found")
+        return False
+
+
+class AssertWebsitesUsesOutdatedPHP(action.CheckAction):
+    last_outdated_version: version.PHPVersion
+    name: str
+    description: str
+
+    def __init__(
+            self,
+            last_outdated_version: str,
+    ):
+        self.name = "checking domains uses outdated PHP"
+        self.last_outdated_version = version.PHPVersion(last_outdated_version)
+        self.description = """We have identified that the domains are using older versions of PHP.
+\tSwitch the following domains to any version higher than {} in order to continue with the conversion process:
+\t- {}
+
+\tYou can achieve this by executing the following command:
+\t> plesk bin domain -u [domain] -php_handler_id plesk-php80-fastcgi
+"""
+
+    def _do_check(self) -> bool:
+        log.debug(f"Checking for minimal PHP version of {self.last_outdated_version}")
+        known_php_versions = get_known_php_versions()
+
+        log.debug(f"Known PHP versions: {known_php_versions}")
+        outdated_php_versions = [php for php in known_php_versions if php < self.last_outdated_version]
+        outdated_php_packages = {f"plesk-php{php.major}{php.minor}": str(php) for php in outdated_php_versions}
+        log.debug(f"Outdated PHP versions: {outdated_php_versions}")
+
+        php_hanlers = {"'{}-fastcgi'", "'{}-fpm'", "'{}-fpm-dedicated'"}
+        outdated_php_handlers = []
+        for outdated_package in outdated_php_packages.keys():
+            outdated_php_handlers += [handler.format(outdated_package) for handler in php_hanlers]
+        log.debug(f"Outdated PHP handlers: {outdated_php_handlers}")
+
+        try:
+            looking_for_domains_sql_request = """
+                SELECT d.name FROM domains d JOIN hosting h ON d.id = h.dom_id WHERE h.php_handler_id in ({});
+            """.format(", ".join(outdated_php_handlers))
+            outdated_php_domains = subprocess.check_output(["/usr/sbin/plesk", "db", looking_for_domains_sql_request],
+                                                           universal_newlines=True)
+            outdated_php_domains = [
+                domain[2:-2] for domain in outdated_php_domains.splitlines()
+                if domain.startswith("|") and not domain.startswith("| name ")
+            ]
+            if len(outdated_php_domains) <= 0:
+                return True
+
+            log.debug(f"Outdated PHP domains: {outdated_php_domains}")
+            outdated_php_domains = "\n\t- ".join(outdated_php_domains)
+            self.description = self.description.format(self.last_outdated_version, outdated_php_domains)
+
+            return False
+        except Exception:
+            log.error("Unable to get domains list from plesk database!")
```

Comment:
Yes, because this issue isn't directly related to our checker, and we can't confirm if any website is using outdated PHP in this case. Actually, we could simply raise an exception again here. However, there might be a problem if a previous tool call has disabled the database, I should check it.